test-network-function/networking/suite.go

```diff
@@ -450,73 +450,66 @@ func listeningPortList(commandlisten []string, nodeOc *interactive.Context, list
 		k.port = p
 		k.protocol = strings.ToUpper(fields[indexprotocolname])
 		k.protocol = strings.ReplaceAll(k.protocol, "\"", "")
-		listeningPorts[k] = ""
+		listeningPorts[k] = true
 	}
 	return nil
 }
 
-func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]string) map[key]string {
-	res := make(map[key]string)
+func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]bool) map[key]bool {
+	res := make(map[key]bool)
 	if len(listeningPorts) == 0 {
 		return res
 	}
 	for k := range listeningPorts {
 		_, ok := declaredPorts[k]
 		if !ok {
-			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
 			res[k] = listeningPorts[k]
 		}
 	}
 	return res
 }
 
 func testListenAndDeclared(env *config.TestEnvironment) {
-	declaredPorts := make(map[key]string)
-	listeningPorts := make(map[key]string)
-	undeclaredPorts := make(map[key]string)
-	var skippedPods []configsections.Pod
-	var failedPods []configsections.Pod
+	var failedcontainers []string
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestUndeclaredContainerPortsUsage)
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
-	OUTER:
-		for _, podUnderTest := range env.PodsUnderTest {
-			for i := 0; i < podUnderTest.ContainerCount; i++ {
-				err := declaredPortList(i, podUnderTest.Name, podUnderTest.Namespace, declaredPorts)
-				if err != nil {
-					tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", i, err, podUnderTest.Name)
-					skippedPods = append(skippedPods, *podUnderTest)
-					continue OUTER
-				}
+		for _, cut := range env.ContainersUnderTest {
+			declaredPorts := make(map[key]bool)
+			listeningPorts := make(map[key]bool)
+			container, _ := strconv.Atoi(cut.ContainerUID)
+			err := declaredPortList(container, cut.PodName, cut.Namespace, declaredPorts)
+			if err != nil {
+				tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", cut.ContainerUID, err, cut.PodName)
+				continue
```

Comment:
1. Change the ClaimFilePrinft message to:
```
tnf.ClaimFilePrintf("Failed to get declared ports list for container %s pod %s ns %s, err %s", cut.ContainerName, cut.PodName, cut.Namespace, err)
```
2. Add this container to the failedContainers slice before "continue".